src/core/mod.rs

```diff
@@ -276,6 +291,70 @@ impl BPFBuilder {
             debug: self.debug,
         })
     }
+
+    #[cfg(any(
+        feature = "v0_18_0",
+        not(feature = "specific"),
+    ))]
+    /// Try constructing a BPF module from the builder
+    pub fn build(mut self) -> Result<BPF, BccError> {
```

Comment:
Is it possible to merge the two build functions again? I suspect it'd be possible to use some `#[cfg(...)]` or `if(cfg!(...))` within the function. It might be nicer than duplicating the function?

Kinda out of scope for this PR, but makes me wonder if we could do something better with the features in general. Not sure there's a good way to do it while also defaulting to the newest bcc version, but it could be cool if we were able to have a `usdt` feature get pulled in as part of `v0_18_0`. If/when we go back and add USDT for older versions, it would just be a matter of bumping bcc-sys dependency and adding the `usdt` feature to `v0_17_0` for example.

Reply:
I think we could probably extract the USDT logic into a helper method which is `cfg`-gated, and then also `cfg`-gate the call to said method.  I'll take a stab at it.

I think I can also take a stab at the thought process here behind USDT support being feature flagged individually.

Reply:
If not, the pattern you're using matches the rest of the codebase. So I'd be fine with it as-is too. Just made me think about the problem and pain points that exist when adding new versions / features.

Reply:
This is better.